tests/PHPStan/Rules/PhpDoc/data/value-of-enum.php

```diff
@@ -0,0 +1,48 @@
+<?php // lint >= 8.1
+
+declare(strict_types=1);
+
+namespace ValueOfEnum;
+
+enum Country: string
+{
+	case NL = 'The Netherlands';
+	case US = 'United States';
+}
+
+enum CountryNo: int
+{
+	case NL = 1;
+	case US = 2;
+}
+
+class Foo {
+	/**
+	 * @param value-of<Country> $countryName
+	 */
+	function hello(string $countryName): void
+	{
+		// ...
+	}
+
+	/**
+	 * @param value-of<Country> $shouldError
+	 */
+	function helloError(int $shouldError): void
+	{
+		// ...
+	}
+	/**
+	 * @param value-of<CountryNo> $shouldError
+	 */
+	function helloError2(string $shouldError): void
+	{
+		// ...
+	}
+
+	function doFoo() {
+		$this->hello(Country::NL);
+		$this->hello(CountryNo::NL);
```

Comment:
this line does not yet error.. but I think this needs a new Rule or enhancing an existing rule..?

should I keep this line or delete it for now?

Reply:
I don't understand. You have a test-case in a PHPDoc incompatibility rule. It's not going to report wrong method calls. For that you need to add a test-case to CallMethodsRuleTest.php.

Reply:
good point. added test-cases in CallMethodsRuleTest